From e8aa4f71d94c84e875dce012da3aadd65d200875 Mon Sep 17 00:00:00 2001
From: Baptiste Mesta <baptiste.mesta@bonitasoft.com>
Date: Sun, 17 Mar 2019 17:21:00 +0100
Subject: [PATCH] fix sp settings

---
 settings.gradle | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/settings.gradle b/settings.gradle
index bfa74b1717..2d3de2727b 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -3,6 +3,22 @@
  */
 
 rootProject.name = 'bonita-modules'
+
+
+apply from: 'community/engine-settings.gradle'
+
+// move all community project in community folder:
+def fixPath
+fixPath = { project ->
+    String relativeProjectPath = project.projectDir.path.replace(settingsDir.path, "")
+    project.projectDir = new File(relativeProjectPath.replace("/engine/", 'community/'))
+    project.children.each fixPath
+}
+rootProject.children.each fixPath
+
+include ':engine'
+project(':engine').projectDir = "$rootDir/community" as File
+
 include(':bonita-engine:bonita-platform:platform-resources')
 include(':bonita-engine:bonita-platform:platform-setup')
 include(':bonita-engine:bonita-platform:platform-setup-test')
@@ -125,3 +141,7 @@ include(':subscription:bonita-integration-tests-sp:bonita-integration-tests-as:b
 include(':subscription:bonita-integration-tests-sp:bonita-integration-tests-as')
 include(':subscription:bonita-integration-tests-sp')
 include(':subscription')
+
+project(':subscription:bpm:bonita-common-sp').projectDir = "$rootDir/subscription/bpm/bonita-common" as File
+project(':subscription:bpm:bonita-client-sp').projectDir = "$rootDir/subscription/bpm/bonita-client" as File
+project(':subscription:bpm:bonita-server-sp').projectDir = "$rootDir/subscription/bpm/bonita-server" as File
-- 
2.20.1

