From 2ff9e17c5e3a617d2f0682ea1fe5db35175e0083 Mon Sep 17 00:00:00 2001
From: Emmanuel Duchastenier <emmanuel.duchastenier@bonitasoft.com>
Date: Thu, 12 Jul 2018 15:56:54 +0200
Subject: [PATCH] do not apply Java plugin to root project Community

---
 community/build.gradle | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/community/build.gradle b/community/build.gradle
index a7f5cb53ff..75dd92f767 100644
--- a/community/build.gradle
+++ b/community/build.gradle
@@ -8,6 +8,8 @@ allprojects {
 }
 
 subprojects {
+    if (it.name == "engine")
+        return
     apply plugin: 'java'
     apply plugin: 'maven-publish'
 
-- 
2.20.1

