From ee2f5baa580653e0fab2ddc77d927aa902ef332f Mon Sep 17 00:00:00 2001
From: Baptiste Mesta <baptiste.mesta@bonitasoft.com>
Date: Thu, 8 Mar 2018 13:09:23 +0100
Subject: [PATCH] run integration test in a specific task

---
 .../bonita-integration-tests-local/build.gradle           | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/community/bonita-integration-tests/bonita-integration-tests-local/build.gradle b/community/bonita-integration-tests/bonita-integration-tests-local/build.gradle
index 33d8788fd7..625b8d1e80 100644
--- a/community/bonita-integration-tests/bonita-integration-tests-local/build.gradle
+++ b/community/bonita-integration-tests/bonita-integration-tests-local/build.gradle
@@ -22,3 +22,11 @@ task testsJar(type: Jar) {
 
 group = 'org.bonitasoft.engine.test'
 publishing.publications.maven.artifact(testsJar)
+
+test {
+    include ''
+}
+
+task integrationTest(type: Test){
+    include 'org/bonitasoft/engine/LocalIntegrationTests.class'
+}
-- 
2.20.1

