From 85e293f70515409528c38533faed375e0b974015 Mon Sep 17 00:00:00 2001
From: Emmanuel Duchastenier <emmanuel.duchastenier@bonitasoft.com>
Date: Thu, 12 Jul 2018 12:07:51 +0200
Subject: [PATCH] javadoc(BOS): generate javadoc for all Community

---
 community/build.gradle | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/community/build.gradle b/community/build.gradle
index 3f66ef1368..f203628682 100644
--- a/community/build.gradle
+++ b/community/build.gradle
@@ -52,4 +52,25 @@ subprojects {
     tasks.withType(JavaCompile) {
         options.encoding = 'UTF-8'
     }
+
+}
+
+
+task allJavadoc(type: Javadoc) {
+    def candidates = [
+            ':platform:platform-setup',
+            ':bpm:bonita-api:bonita-common-api',
+            ':bpm:bonita-api:bonita-common-api-impl',
+            ':bpm:bonita-util:bonita-client-util'
+    ]
+    source candidates.collect { project(it).sourceSets.main.allJava }
+    classpath = files(candidates.collect { project(it).sourceSets.main.compileClasspath })
+    destinationDir = buildDir
+    failOnError = false
+    title = "Bonita Engine ${version} API"
+    options.links("https://docs.oracle.com/javase/8/docs/api/")
+    // TODO change it when SP is up and running:
+    options.stylesheetFile = new File("$projectDir/../subscription/bpm/bonita-global-javadoc/src/main/javadoc", "stylesheet.css")
+    options.overview = new File("$projectDir/../subscription/bpm/bonita-global-javadoc/src/main/javadoc", "overview.html")
+    options.bottom = "Copyright &copy; ${new Date().format('yyyy')} <a href=\"http://community.bonitasoft.com/\">Bonitasoft S.A.</a> All rights reserved."
 }
-- 
2.20.1

