From cef8cc18bffa7a337cdfd20323135b7f553bf72e Mon Sep 17 00:00:00 2001
From: Emmanuel Duchastenier <emmanuel.duchastenier@bonitasoft.com>
Date: Fri, 13 Jul 2018 10:42:19 +0200
Subject: [PATCH] Fix root build.gradle about maven repo

---
 build.gradle | 34 +++++++++++++++-------------------
 1 file changed, 15 insertions(+), 19 deletions(-)

diff --git a/build.gradle b/build.gradle
index 75db969885..f13adf1c29 100644
--- a/build.gradle
+++ b/build.gradle
@@ -1,26 +1,22 @@
-allprojects  {
-  apply plugin: 'maven'
+allprojects {
+    apply plugin: 'maven'
 
-  group = 'org.bonitasoft'
-version = '7.8.0-SNAPSHOT'
+    group = 'org.bonitasoft'
+    version = '7.8.0-SNAPSHOT'
 }
 
 subprojects {
-  apply plugin: 'java'
-  sourceCompatibility = 1.8
-  targetCompatibility = 1.8
+    apply plugin: 'java'
+    sourceCompatibility = 1.8
+    targetCompatibility = 1.8
 
-  
-  repositories {
-    mavenLocal()
-    
-    maven { url "http://dev.rd.lan/userContent/m2_repo/" }
-    maven { url "http://repositories.rd.lan/maven/all/" }
-    maven { url "http://repo.maven.apache.org/maven2" }
-    maven { url "https://oss.sonatype.org/content/repositories/snapshots" }
-  }
+    repositories {
+        mavenLocal()
+        maven { url "http://repositories.rd.lan/maven/releases/" }
+        maven { url "http://repositories.rd.lan/maven/snapshots/" }
+        maven { url "http://repositories.rd.lan/maven/all/" }
+        maven { url "https://oss.sonatype.org/content/repositories/snapshots" }
+        mavenCentral()
+    }
 
-  
-  
-  
 }
-- 
2.17.1

