From 27cdadefe8a92fb9f38b242192e0499ebdf04243 Mon Sep 17 00:00:00 2001
From: Baptiste Mesta <baptiste.mesta@bonitasoft.com>
Date: Mon, 16 Jul 2018 17:26:12 +0200
Subject: [PATCH] add JVM flags in cluster tests

---
 .../com/bonitasoft/engine/NodeLauncher.java   | 46 +++++++++++++------
 1 file changed, 32 insertions(+), 14 deletions(-)

diff --git a/subscription/bonita-integration-tests-sp/bonita-integration-tests-as/bonita-integration-cluster/src/test/java/com/bonitasoft/engine/NodeLauncher.java b/subscription/bonita-integration-tests-sp/bonita-integration-tests-as/bonita-integration-cluster/src/test/java/com/bonitasoft/engine/NodeLauncher.java
index 4928b93221..ce6ab9af81 100644
--- a/subscription/bonita-integration-tests-sp/bonita-integration-tests-as/bonita-integration-cluster/src/test/java/com/bonitasoft/engine/NodeLauncher.java
+++ b/subscription/bonita-integration-tests-sp/bonita-integration-tests-as/bonita-integration-cluster/src/test/java/com/bonitasoft/engine/NodeLauncher.java
@@ -18,8 +18,12 @@ import java.io.IOException;
 import java.io.InputStream;
 import java.io.InputStreamReader;
 import java.nio.file.Paths;
+import java.lang.management.ManagementFactory;
+import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 import org.bonitasoft.engine.api.ApiAccessType;
 import org.bonitasoft.engine.util.APITypeManager;
@@ -33,24 +37,38 @@ public class NodeLauncher {
     public static Process start(String nodeName, String databaseUrl, int port, int healthCheckServerPort, String clusterName) throws IOException {
         File target = new File("target");
 
-        ProcessBuilder processBuilder = new ProcessBuilder(getJavaPath(),
-                "-Xms64m",
-                "-Xmx320m",
-                "-Dsysprop.bonita.cluster=true",
-                "-Dbonita.client.home=" + System.getProperty("bonita.client.home"),
-                "-Dsysprop.bonita.cluster.group.name="+clusterName,
-                "-Ddb.url=" + databaseUrl,
-                "-D" + NODE_NAME + "=" + nodeName,
-                "-cp",
-                System.getProperty("java.class.path"),
-                JVMNode.class.getName(),
-                String.valueOf(port),
-                String.valueOf(healthCheckServerPort)
-        );
+
+        List<String> command = new ArrayList<>();
+        command.add(getJavaPath());
+        List<String> jvmFlags = ManagementFactory
+                .getRuntimeMXBean()
+                .getInputArguments()
+                .stream()
+                .filter(arg -> arg.startsWith("-XX"))
+                .collect(Collectors.toList());
+        if (!jvmFlags.isEmpty()) {
+            System.out.println("Adding flags from source JVM: " + jvmFlags);
+            command.addAll(jvmFlags);
+        }
+        command.add("-Dsysprop.bonita.cluster=true");
+        command.add("-Dbonita.client.home=" + System.getProperty("bonita.client.home"));
+        command.add("-Dbonita.cluster.group.name="+clusterName);
+        command.add("-Ddb.url=" + databaseUrl);
+        command.add("-D" + NODE_NAME + "=" + nodeName);
+        command.add("-cp");
+        command.add(System.getProperty("java.class.path"));
+        command.add(JVMNode.class.getName());
+        command.add(String.valueOf(port));
+        command.add(String.valueOf(healthCheckServerPort));
+
+        ProcessBuilder processBuilder = new ProcessBuilder(command);
+
+
         Process process = processBuilder
                 .directory(getNodeDir(nodeName, target))
                 .start();
 
+        System.out.println("Starting node [" + nodeName + "]: " + processBuilder.command());
         printLogsOfFork(nodeName, process);
 
         setConnectionPort(port);
-- 
2.20.1

