From 7bd50f226be3ed81181152ba5eaffb577e187d5e Mon Sep 17 00:00:00 2001
From: Baptiste Mesta <baptiste.mesta@bonitasoft.com>
Date: Fri, 13 Jul 2018 17:17:13 +0200
Subject: [PATCH] fix integration tests

---
 .../bonita-integration-cluster/build.gradle                | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/subscription/bonita-integration-tests-sp/bonita-integration-tests-as/bonita-integration-cluster/build.gradle b/subscription/bonita-integration-tests-sp/bonita-integration-tests-as/bonita-integration-cluster/build.gradle
index 9fb6022dfc..04c122761a 100644
--- a/subscription/bonita-integration-tests-sp/bonita-integration-tests-as/bonita-integration-cluster/build.gradle
+++ b/subscription/bonita-integration-tests-sp/bonita-integration-tests-as/bonita-integration-cluster/build.gradle
@@ -18,4 +18,11 @@ test {
   include ''
 }
 task integrationTest(type: Test) {
+  include 'com/bonitasoft/engine/ClusteredTestSuiteIT.class'
+  def testDir = new File(buildDir, "test")
+  workingDir = testDir
+  doFirst {
+    testDir.mkdirs()
+    new File(testDir, "target").mkdir()
+  }
 }
-- 
2.20.1

