From 0542435676d87021d10625570877fdc6b0e21eed Mon Sep 17 00:00:00 2001
From: Baptiste Mesta <baptiste.mesta@bonitasoft.com>
Date: Fri, 13 Jul 2018 16:35:56 +0200
Subject: [PATCH] update jenkins file

---
 Jenkinsfile | 13 ++-----------
 1 file changed, 2 insertions(+), 11 deletions(-)

diff --git a/Jenkinsfile b/Jenkinsfile
index f006f6520fc..b2f73529646 100644
--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -17,20 +17,11 @@ timestamps {
             }
 
             try {
-                if (isMasterOrDev) {
-                    slackStage('Build and Publish') {
-                        mvn "clean deploy -Ptests,javadoc -DaltDeploymentRepository=${env.ALT_DEPLOYMENT_REPOSITORY_SNAPSHOTS}"
-                    }
-                } else {
                     stage('Build and Test') {
-                        mvn 'install -Ptests'
-                    }
-                }
-                stage('AS Tests') {
-                    mvn 'verify -Ptests,as-tests -pl :bonita-integration-tests-as,:bonita-integration-tests-as-sp -amd'
+                     sh './gradlew build iT'
                 }
             } finally {
-                junit '**/target/*-reports/*.xml'
+                junit '**/build/test-results/**/*.xml'
             }
         }
     }
-- 
2.17.1

