From c2e1a97c732859cfcd17c18e19868a39cd5b46a9 Mon Sep 17 00:00:00 2001
From: Baptiste Mesta <baptiste.mesta@gmail.com>
Date: Fri, 14 Sep 2018 23:27:25 +0200
Subject: [PATCH] publish non shaded bdm generator for studio

---
 .../bonita-business-data-generator/build.gradle           | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/community/services/bonita-business-data/bonita-business-data-generator/build.gradle b/community/services/bonita-business-data/bonita-business-data-generator/build.gradle
index 133c97b091..c1f288ecd7 100644
--- a/community/services/bonita-business-data/bonita-business-data-generator/build.gradle
+++ b/community/services/bonita-business-data/bonita-business-data-generator/build.gradle
@@ -23,3 +23,11 @@ dependencies {
 
 group = 'org.bonitasoft.engine.data'
 description = 'Business Data Jar Generator'
+
+publishing {
+    publications {
+        mavenJava(MavenPublication) {
+            from project.components.java
+        }
+    }
+}
-- 
2.20.1

