From 5a65fb7074433138ce92ca4f0738191053c9cbb8 Mon Sep 17 00:00:00 2001
From: Baptiste Mesta <baptiste.mesta@gmail.com>
Date: Sun, 16 Sep 2018 18:34:26 +0200
Subject: [PATCH] publish bonita-integration-test-client (selenium)

---
 .../bonita-integration-tests-client/build.gradle           | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/community/bonita-integration-tests/bonita-integration-tests-client/build.gradle b/community/bonita-integration-tests/bonita-integration-tests-client/build.gradle
index 3f3624ef3a..104e294775 100644
--- a/community/bonita-integration-tests/bonita-integration-tests-client/build.gradle
+++ b/community/bonita-integration-tests/bonita-integration-tests-client/build.gradle
@@ -46,3 +46,10 @@ test.doFirst {
         rename 'bonita-client-invalid.properties', 'bonita-client-community.properties'
     }
 }
+publishing {
+    publications {
+        mavenJava(MavenPublication) {
+            from project.components.java
+        }
+    }
+}
-- 
2.20.1

