From 6137e74f22d14a47ae1a13d7b9423dbc03aa214a Mon Sep 17 00:00:00 2001
From: Emmanuel Duchastenier <emmanuel.duchastenier@bonitasoft.com>
Date: Mon, 16 Jul 2018 15:31:00 +0200
Subject: [PATCH] Fix Java/Groovy compilation encoding

---
 build.gradle           | 4 ++++
 community/build.gradle | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/build.gradle b/build.gradle
index 17b1f3ef96..d7d61d2f28 100644
--- a/build.gradle
+++ b/build.gradle
@@ -53,5 +53,9 @@ subprojects {
                 }
             }
         }
+
+        tasks.withType(AbstractCompile) {
+            options.encoding = 'UTF-8'
+        }
     }
 }
diff --git a/community/build.gradle b/community/build.gradle
index 4cfb7d0b48..a66ea9c7dc 100644
--- a/community/build.gradle
+++ b/community/build.gradle
@@ -13,6 +13,7 @@ subprojects {
     apply plugin: 'java'
     apply plugin: 'maven-publish'
 
+
     repositories {
         mavenLocal()
         maven {
@@ -63,6 +64,9 @@ subprojects {
                 }
             }
         }
+        tasks.withType(AbstractCompile) {
+            options.encoding = 'UTF-8'
+        }
     }
     tasks.withType(JavaCompile) {
         options.encoding = 'UTF-8'
-- 
2.20.1

